pulp_maven/app/maven_deploy_api.py

```diff
@@ -77,9 +77,12 @@ class MavenApiViewSet(APIView):
     authentication_classes = []
     permission_classes = []
 
-    def redirect_to_content_app(self, distribution, relative_path):
+    def redirect_to_content_app(self, distribution, relative_path, request):
+        scheme = request.scheme
```

Comment:
This is probably right, but also this could be incorrect because the gateway may be reverse proxying with http while the client to the gateway uses https. I think you also want to get that data from a header too X-Forwarded-Proto

I think maybe django eats that data up and uses it in request.scheme, but it's also dependent (I believe) on settings so you could easily mess up this feature with django settings.

Reply:
good point. i'll update the PR.